### ISSUE TITLE
Add conditional formats

### ISSUE THREAD

**Issue by eduardoboucas** [dadi/cdn#400]
CDN could conditionally deliver images on certain formats based on whether they are supported by the requesting client, as long as a fallback is supplied. For example, `?format=webp,jpg` could check for WebP support in the `Accept` header and use it if supported, delivering a JPEG otherwise.

**Comment by elledienne** [comment]
@eduardoboucas why not something like `?format=auto`? 

I believe the point of such a feature is to push the responsibility of choosing the best format to the CDN abstracting away any complexity (no need to know about `webp` and no need to update parameters if a new standard comes out).

**Comment by eduardoboucas** [comment]
Hi @elledienne, thanks for sharing that.

I'm with you 100%. Giving CDN the capability to decide various things on behalf of the user is a route we're pursuing and this is just a step in that direction.

An `auto` format would probably need to take more things into account than simply the `Accept` header to decide what format to use. I'd also like for it to be able to determine the best compression level and other parameters.

In the meantime, this is a simple and non-breaking way of making WebP output more useful: giving users the ability to choose a fallback.

TL;DR: stay tuned 😄

**Comment by elledienne** [comment]
@eduardoboucas good to hear that that’s the plan!
Do you already have a roadmap for this feature or planned how to implement that?
